impl/src/main/java/io/jsonwebtoken/impl/DefaultClaims.java

```diff
@@ -22,6 +22,13 @@
 
 public class DefaultClaims extends JwtMap implements Claims {
 
+    private static final String CONVERSION_ERROR_MSG = "Cannot convert existing claim value of type '%s' to desired type " +
+            "'%s'. JJWT only converts simple String, Date, Long, Integer, Short and Byte types automatically. " +
+            "Anything more complex is expected to be already converted to your desired type by the JSON Deserializer " +
+            "implementation. You may specify a custom Deserializer for a JwtParser with the desired conversion " +
+            "configuration via the JwtParserBuilder.deserializeJsonWith() method. " +
+            "See https://github.com/jwtk/jjwt#custom-json-processor for more information.";
```

Comment:
I think we should add one more sentence at the end of this paragraph:

"If using Jackson, you can specify custom claim POJO types as described in <url-here>."

Where <url-here> is of course the section in the documentation that explains how to do this that you're presumably writing in #500 

Since I would think Jackson will be the most widely used parser.

Thoughts?